packages/file_selector/file_selector_android/android/src/main/java/io/flutter/plugins/file_selector/FileSelectorPlugin.java

```diff
@@ -99,24 +104,41 @@ private void tearDown() {
     }
   }
 
+  @RequiresApi(api = Build.VERSION_CODES.LOLLIPOP)
   @Override
   public void onMethodCall(@NonNull MethodCall call, @NonNull MethodChannel.Result rawResult) {
     if (activityState == null || activityState.getActivity() == null) {
       rawResult.error("no_activity", "file_selector plugin requires a foreground activity.", null);
       return;
     }
+
+    final HashMap arguments = (HashMap) call.arguments;
+
     MethodChannel.Result result = new MethodResultWrapper(rawResult);
     FileSelectorDelegate delegate = activityState.getDelegate();
+    delegate.clearCache();
```

Comment:
I know we have to clear the cache, but why here precisely?

Reply:
I hadn't noticed that there was a tearDown, I have moved this there. Thanks!